Package/CreatePackage/Package.cs

```diff
@@ -673,10 +680,57 @@ private static void UpdateVersionInfo(string tempDir, List<PackageFile> files, S
             log.Info(timer,"Updated assembly version info using Mono method.");
         }
 
+        private static bool VerifyArchiveIntegrity(FileStream fs)
+        {
+            log.Debug($"Verifying package integrity.");
+            // Ensure the stream is fully written to the disk
+            fs.Flush();
+
+            {   // Verify that we can read the package definition
+                try
+                {
+                    PackageDef.FromPackage(fs.Name);
```

Comment:
It seems this now consistently fails. That is probably because the file stream is open. 

But this should be fixable with some FileShare settings inside FromPackage.

Reply:
Argh, I only tested on Linux!